submissions/ses3332002/tiny-js-world/index.js

```diff
@@ -0,0 +1,98 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+const dog = {
+  species: "dog",
+  name: "Toby",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Woof-woof!",
+  friends: [
+    "Puss"
+  ]
+};
+
+const cat = {
+  species: "cat",
+  name: "Puss",
+  gender: "male",
+  legs: 4,
+  hands: 0,
+  saying: "Meow!",
+  friends: [
+    "Toby",
+    "Halley"
+  ]
+};
+
+const man = {
+  species: "human",
+  name: "Jack",
+  gender: "male",
+  legs: 2,
+  hands: 2,
+  saying: "Hey baby!",
+  friends: [
+    "Jennie",
+    "Halley"
+  ]
+};
+
+const woman = {
+  species: "human",
+  name: "Jennie",
+  gender: "female",
+  legs: 2,
+  hands: 2,
+  saying: "Hi darling!",
+  friends: [
+    "Jack",
+    "Toby"
+  ]
+};
+
+const catWoman = {
+  species: "super-hero",
+  name: "Halley",
+  gender: "female",
+  legs: 2,
+  hands: 2,
+  saying: cat.saying,
+  friends: [
+    "Puss"
+  ]
+};
+
+const inhabitants = [dog, cat, man, woman, catWoman];
+const propsToPrint = ["species", "name", "gender", "legs", "hands", "friends", "saying"];
+
+inhabitants.forEach(item => {
+  print(Object.entries(item).filter(entry => propsToPrint.includes(entry[0])).map(entry => Array.isArray(entry[1]) ? entry[1].join(', ') : entry[1]).join('; '));
```

Comment:
You are not required to fix it here. Just consider emplying formatting to improve readability.
E.g.
```
array
  .filter(..)
  .map(...)
  .join(...)
```

Consider using [prettier](https://prettier.io/) to do this job for you. Some of the IDEs offer on-the-fly formatting.

Here `entry[0]` and `entry[1]` contain values for entities that are different by nature. Use destructuring to have named variables instead of cryptic referencing by index, thus improving readability. E.g. `Object.entries(...).filter(([key, value]) => ... ...)`

You may also skip an unused part of an entry. E.g.
`Object.entries(...).map(([, value]) => ... .isArray(value) ...)`

Reply:
I remade it.